test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/IntegrationTestBase.cs

```diff
@@ -229,5 +224,98 @@ protected void AssertSourceMappingsMatchBaseline(RazorCodeDocument document)
+
+        private static string NormalizeNewLines(string content)
+        {
+            return Regex.Replace(content, "(?<!\r)\n", "\r\n", RegexOptions.None, TimeSpan.FromSeconds(10));
+        }
+
+        private class IntegrationTestProjectEngine : DefaultRazorProjectEngine
+        {
+            public IntegrationTestProjectEngine(
+                RazorProjectEngine innerEngine)
+                : base(innerEngine.Configuration, innerEngine.Engine, innerEngine.FileSystem, innerEngine.ProjectFeatures)
+            {
+            }
+
+            protected override void ProcessCore(RazorCodeDocument codeDocument)
+            {
+                // This will ensure that we're not putting any randomly generated data in a baseline.
+                codeDocument.Items[CodeRenderingContext.SuppressUniqueIds] = "test";
+
+                // This is to make tests work cross platform.
+                codeDocument.Items[CodeRenderingContext.NewLineString] = "\r\n";
+
+                base.ProcessCore(codeDocument);
+            }
+        }
+
+        private class IntegrationTestImportFeature : RazorProjectEngineFeatureBase, IImportProjectFeature
+        {
+            private Assembly _assembly;
+            private IImportProjectFeature _existingImportFeature;
+
+            public IntegrationTestImportFeature(Assembly assembly, IImportProjectFeature existingImportFeature)
+            {
+                _assembly = assembly;
+                _existingImportFeature = existingImportFeature;
+            }
+
+            protected override void OnInitialized()
+            {
+                _existingImportFeature.ProjectEngine = ProjectEngine;
+            }
+
+            public IReadOnlyList<RazorProjectItem> GetImports(RazorProjectItem projectItem)
+            {
+                var imports = new List<RazorProjectItem>();
+
+                while (true)
+                {
+                    var importsFileName = Path.ChangeExtension(projectItem.FilePathWithoutExtension + "_Imports" + imports.Count.ToString(), ".cshtml");
+                    var importsFile = TestFile.Create(importsFileName, _assembly);
+                    if (!importsFile.Exists())
+                    {
+                        break;
+                    }
+
+                    var importContent = importsFile.ReadAllText();
+                    var normalizedContent = NormalizeNewLines(importContent);
+                    var importItem = new TestRazorProjectItem(importsFileName)
+                    {
+                        Content = normalizedContent
+                    };
+                    imports.Add(importItem);
+                }
+
+                imports.AddRange(_existingImportFeature.GetImports(projectItem));
+
+                return imports;
+            }
+        }
+
+        private class IntegrationTestFileSystem : RazorProjectFileSystem
+        {
+            public static IntegrationTestFileSystem Default = new IntegrationTestFileSystem();
```

Comment:
Isn't htis the same as empty?

Reply:
Empty still expects the file path to start with `/`. We don't do that in our tests.